src/_guides/language/effective-dart/usage_migrated.md

```diff
@@ -172,6 +240,100 @@ preferred for three main reasons:
     expression is `null`. With `== true`, you have to think through the boolean
     logic to realize that means that a `null` gets converted to *false*.
 
+**Exception:** Using a null-aware operator on a variable inside a condition does
+not promote the variable to a non-nullable type. If you want the variable to be
+promoted inside the body of the if statement, it may be better to use an
+explicit `!= null` check instead of `?.` followed by `??`:
```

Comment:
```suggestion
**Exception:** Using a null-aware operator on a variable inside a condition
doesn't promote the variable to a non-nullable type. If you want the variable
to be promoted inside the body of the `if` statement, it might be better to use
an explicit `!= null` check instead of `?.` followed by `??`:
```

For the may -> might change, see https://developers.google.com/style/word-list#letter-m.

I have to point out that these diffs would've been easier to read if we were using semantic line breaks. (I'll try not to proselytize any more!)

Reply:
I know semantic line breaks are better. You're absolutely right. They just hurt my soul to look at. I'm sorry.

Reply:
It helps to think of them as free-form poetry. (I'm probably using that terminology wrong; @RedBrogdon would know the real poetical terms.)